libs/labelbox/src/labelbox/schema/workflow/workflow.py

```diff
@@ -0,0 +1,1504 @@
+            self._edges_cache = None
+            # For non-ValueError exceptions, wrap them with a more informative message
+            logger.error(f"Error updating workflow: {e}")
+            raise ValueError(f"Failed to update workflow: {e}") from e
+
+    def _prepare_config_for_api(self) -> Dict[str, Any]:
+        """Prepare the workflow configuration for saving to the API.
+
+        Filters down node and edge attributes to only those accepted by the API.
+
+        Returns:
+            Dict[str, Any]: The cleaned configuration ready for API submission
+        """
+        # Make sure we include only fields that the API accepts
+        clean_config = {
+            "nodes": [
+                {
+                    key: value
+                    for key, value in {
+                        "id": node["id"],
+                        "position": node["position"],
+                        "definitionId": node["definitionId"],
+                        "label": node.get("label"),
+                        "filterLogic": node.get("filterLogic"),
+                        "customFields": node.get("customFields"),
+                        "config": node.get("config"),
+                        "filters": self._serialize_filters(node.get("filters")),
+                    }.items()
+                    if value is not None
+                }
+                for node in self.config.get("nodes", [])
+            ],
+            "edges": [
+                {
+                    "id": edge["id"],
+                    "source": edge["source"],
+                    "target": edge["target"],
+                    "sourceHandle": edge["sourceHandle"],
+                    "targetHandle": edge["targetHandle"],
+                }
+                for edge in self.config.get("edges", [])
+            ],
+        }
+        return clean_config
+
+    def _serialize_filters(self, filters):
+        """Serialize filters to ensure they are JSON-serializable.
+
+        Args:
+            filters: The filters value which could be a list, ProjectWorkflowFilter object, or None
+
+        Returns:
+            List or None: JSON-serializable filters or None
+        """
+        if filters is None:
+            return None
+
+        # If it's a ProjectWorkflowFilter object, convert it to a list
+        if hasattr(filters, "to_dict") and callable(filters.to_dict):
+            try:
+                return filters.to_dict()
+            except Exception:
+                # If to_dict() fails, try to access filters attribute directly
+                if hasattr(filters, "filters") and isinstance(
+                    filters.filters, list
+                ):
+                    return filters.filters
+
+        # If it's already a list, return as is
+        if isinstance(filters, list):
+            return filters
+
+        # For any other type, return None to avoid serialization errors
+        return None
+
+    def reset_config(self) -> "ProjectWorkflow":
```

Comment:
nit: We might want to consider creating the default nodes as part of this process - I'll leave it up to you.